### ISSUE TITLE
Find my downloaded files in the browser

### ISSUE THREAD

**Issue by boek** [mozilla-mobile/fenix#349]
**Part of Meta: #4417**

### User Story
As a user, I want to be able to easily and quickly find and access my downloaded files from within the browser, so I don’t have to look for them in my Android file system.

### Requirements
Populate the downloads UI with a list of files that have been downloaded

(for the v1 minimum, let's copy what fennec does)

### Acceptance Criteria
* [ ] I can navigate to the Downloads section in my browser
* [ ] I can view the list of my COMPLETED downloaded files in the browser
* [ ] There is a cap on how long we display a download
* [ ] If user deletes a downloaded file from the file system, I no longer see the download in the browser 
* [ ] I can tap on a downloaded file and open it using the supported Android system app

**Comment by nahuhh** [comment]
To add to this request
- a ui for the downloader (within menu > library) where you can view current and past downloads, along with the ability to pause/resume/cancel ones that are in progress

**Comment by djc** [comment]
As a quick fix, even a menu item that sends me to the Files app would be an improvement over the current situation, where downloads basically completely disappear after the initial notification.

**Comment by sblatz** [comment]
Closing in favor of #4417.

**Comment by sblatz** [comment]
@vesta0 The download feature currently allows users to open the downloaded file in the respective app (pdf opens in a PDF viewer, etc.). Is there something different we're wanting to do here?

**Comment by vesta0** [comment]
This feature is just about being able to view the files user has previously downloaded using Fenix. Opening the files, as you said, is already possible using the respective apps. 

Note that this feature has been de-prioritized for Q1

**Comment by liuche** [comment]
To get started. we can add this under Advanced, and then figure out where it lives later.

**Comment by liuche** [comment]
from sprint planning, could we get some feedback from you @brampitoyo about where this item could live?

**Comment by brampitoyo** [comment]
@liuche This “Downloads” menu item should definitely appear next to the “History, Bookmarks, Synchronised tabs” set of menu items.

For now, let’s make it appear between “History” and “Add-ons”.

**Comment by AmyYLee** [comment]
Removed needs:UX-feedback since it looks like UX has answered this question.

**Comment by Mugurell** [comment]
When reviewing this I saw we don't yet have an icon for the new menu option.
And also there are no specs for how a download item should appear in the list. Should we use 1-1 the layout Fennec uses?

**Comment by brampitoyo** [comment]
I think this may be @apbitner’s area.

I may be able to unblock you here, although my answer isn’t final:

- The icon we use for our menu item should be the same Download icon that we’ve used elsewhere in the app (e.g. Download modal bottom sheet, notification shade)
- The Downloads page itself may be based off Fennec’s original design, but it would have to be styled like other Fenix in-app pages (History, Bookmarks, Synced Tabs)

**Comment by liuche** [comment]
@kglazko will create a new frontend-engineering ticket for the v1 of the View Downloads work and add it to the board.

This ticket will be tracking to UX work (and will become an eng ticket once we get mocks).

**Comment by liuche** [comment]
UX will provide icons by next Wednesday, but in the meantime, we can do the default icons. We want to ship with the final icons though.

**Comment by kglazko** [comment]
This was merged with Nightly feature flags and a QA-needed label has been added.

**Comment by liuche** [comment]
For QA: the downloads won't persist until this AC issue has landed: https://github.com/mozilla-mobile/android-components/issues/7762

**Comment by sflorean** [comment]
Testing results on Nightly 8/20:
-  I can navigate to the Downloads section in my browser - **verified as fixed** 
-  I can view the list of my COMPLETED downloaded files in the browser - failed downloads aren't listed, but if we cancel the download they are displayed in the Downloads section (filed #13983)
-  There is a cap on how long we display a download -  do we have here some measurements/extra details?
- If user deletes a downloaded file from the file system, I no longer see the download in the browser - **files are still displayed**, filed #13979
- I can tap on a downloaded file and open it using the supported Android system app  - **verified as fixed**, but for long tap the browser crashes (#13974 is filed).

cc @kglazko

**Comment by Amejia481** [comment]
For testing this feature:

You will be able to :

1. Navigate to the Downloads section.
2. View the list of my COMPLETED downloaded files.
3. Tap on a downloaded file and open it using the supported Android system app.
4. Downloaded files should be persistent even after the application gets restarted. 

To test that  4 (downloads are persistent), after downloaded a file, close the app using the task manager and re open the app, then go to the three dots menu -> Downloads and downloaded files should be shown.

**Comment by liuche** [comment]
"If user deletes a downloaded file from the file system, I no longer see the download in the browser" is not implemented yet, and will not be present in this build.

**Comment by sv-ohorvath** [comment]
> "If user deletes a downloaded file from the file system, I no longer see the download in the browser" is not implemented yet, and will not be present in this build.

Verified as fixed on Beta 81.1.0, including this: the files deleted from the system file manager will be removed from the browser Downloads list as well (fixed in https://github.com/mozilla-mobile/fenix/issues/13979).
